internal/elasticsearch/transform/transform.go

```diff
@@ -0,0 +1,548 @@
+	}
+
+	timeout, err := time.ParseDuration(d.Get("timeout").(string))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	params.Timeout = timeout
+
+	params.Enabled = d.Get("enabled").(bool)
+
+	if diags := elasticsearch.UpdateTransform(ctx, client, updatedTransform, &params); diags.HasError() {
+		return diags
+	}
+
+	return resourceTransformRead(ctx, d, meta)
+}
+
+func resourceTransformDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
+	client, diags := clients.NewApiClient(d, meta)
+	if diags.HasError() {
+		return diags
+	}
+
+	id := d.Id()
+	compId, diags := clients.CompositeIdFromStr(id)
+	if diags.HasError() {
+		return diags
+	}
+
+	if diags := elasticsearch.DeleteTransform(ctx, client, &compId.ResourceId); diags.HasError() {
+		return diags
+	}
+
+	return diags
+}
+
+func getTransformFromResourceData(ctx context.Context, d *schema.ResourceData, name string) (*models.Transform, error) {
+
+	var transform models.Transform
+	transform.Name = name
+
+	if v, ok := d.GetOk("description"); ok {
+		transform.Description = v.(string)
+	}
+
+	if v, ok := d.GetOk("source"); ok {
+		definedSource := v.([]interface{})[0].(map[string]interface{})
+
+		transform.Source = new(models.TransformSource)
+		indices := make([]string, 0)
+		for _, i := range definedSource["indices"].([]interface{}) {
+			indices = append(indices, i.(string))
+		}
+		transform.Source.Indices = indices
+
+		if v, ok := definedSource["query"]; ok && len(v.(string)) > 0 {
+			var query interface{}
+			if err := json.NewDecoder(strings.NewReader(v.(string))).Decode(&query); err != nil {
+				return nil, err
+			}
+			transform.Source.Query = query
+		}
+
+		if v, ok := definedSource["runtime_mappings"]; ok && len(v.(string)) > 0 {
+			var runtimeMappings interface{}
+			if err := json.NewDecoder(strings.NewReader(v.(string))).Decode(&runtimeMappings); err != nil {
+				return nil, err
+			}
+			transform.Source.RuntimeMappings = runtimeMappings
+		}
+	}
+
+	if v, ok := d.GetOk("destination"); ok {
+
+		definedDestination := v.([]interface{})[0].(map[string]interface{})
+		transform.Destination = new(models.TransformDestination)
+
+		transform.Destination.Index = definedDestination["index"].(string)
+
+		if pipeline, ok := definedDestination["pipeline"]; ok && len(pipeline.(string)) > 0 {
+			transform.Destination.Pipeline = pipeline.(string)
+		}
+	}
+
+	if v, ok := d.GetOk("pivot"); ok {
+		var pivot interface{}
+		if err := json.NewDecoder(strings.NewReader(v.(string))).Decode(&pivot); err != nil {
+			return nil, err
+		}
+		transform.Pivot = pivot
+	}
+
+	if v, ok := d.GetOk("latest"); ok {
+		var latest interface{}
+		if err := json.NewDecoder(strings.NewReader(v.(string))).Decode(&latest); err != nil {
+			return nil, err
+		}
+		transform.Latest = latest
+	}
+
+	if v, ok := d.GetOk("frequency"); ok {
+		transform.Frequency = v.(string)
+	}
+
+	if v, ok := d.GetOk("metadata"); ok {
+		metadata := make(map[string]interface{})
```

Comment:
I think this should be fine as a var declaration?

```suggestion
		var metadata map[string]interface{}
```